### PR TITLE
Tweaks to asset sharing to support Unreal implementation

### PR DESCRIPTION
**Pull request by kring**
This is a PR into #926. It goes with CesiumGS/cesium-unreal#1527.

### CODE REVIEW

File: CesiumGltf/include/CesiumGltf/AttributeSemantics.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include <string>
+
+namespace CesiumGltf {
+
+/**
+ * @brief The standard glTF attribute semantics from the specification.
+ */
+struct AttributeSemantics {
+  /**
+   * @brief Unitless XYZ vertex positions.
```
Comment (j9liu) [inline CesiumGltf/include/CesiumGltf/AttributeSemantics.h:13]: Just a drive by comment, I think the glTF spec says the [linear distances](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#coordinate-system-and-units) are in meters? 😅 

image

Comment (kring) [reply CesiumGltf/include/CesiumGltf/AttributeSemantics.h:13]: I copied "Unitless XYZ vertex positions" right out of the glTF specification. See the table in this section:
https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#meshes-overview

I think the documentation is not actually contradicting itself. The POSITION semantic is unitless, but after it's transformed by the Node hierarchy it's supposed to be meters.

**Comment by lilleyse** [comment]
Couple questions since I have a similar enum/struct thing in another project and would like to use this eventually

* Should it be called `VertexAttributeSemantics` in case we add `InstanceAttributeSemantics` in the future?
* Is there the possibility of allowing custom semantics like `_FEATURE_ID_n`?

**Comment by kring** [comment]
> Should it be called VertexAttributeSemantics in case we add InstanceAttributeSemantics in the future?

Good idea, I'll rename it.

> Is there the possibility of allowing custom semantics like _FEATURE_ID_n?

Do you have an idea what that would look like? It's not obvious to me what value we could provide for custom attributes in a class like this, but I could be missing something.

**Comment by kring** [comment]
Oh, ok, I don't have any problem with adding extra commonly-used attributes from extensions to this class. I thought you were suggesting we make it externally-extensible somehow.

**Pull request merged**